### PR TITLE
add arm64 support for M1 Mac

### PR DESCRIPTION
**Pull request by DanboDuan**

### CODE REVIEW

File: cmake/modules/SwiftSupport.cmake
Comment (compnerd) [inline cmake/modules/SwiftSupport.cmake:12]: Similar to swift-system, not sure if the `AArch64` case is needed, but this seems fine.

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please test

File: cmake/modules/SwiftSupport.cmake
Comment (DanboDuan) [reply cmake/modules/SwiftSupport.cmake:12]: I can delete `AArch64 ` if not needed. the code is just copied from swift-atomics/cmake/modules/SwiftSupport.cmake. Better to keep it

Comment (tomerd) [reply cmake/modules/SwiftSupport.cmake:12]: I think both are needed to cover apple silicon and arm linux. cc @drexin

Comment (compnerd) [reply cmake/modules/SwiftSupport.cmake:12]: @tomerd I'm not sure - I'm talking about _literally_ `AArch64`, not `aarch64`.  Its a question of is there an extra case being matched against or not.  Either way, it doesn't actually matter,

Comment (drexin) [reply cmake/modules/SwiftSupport.cmake:12]: I’m not sure about this particular case, but I have seen AMD64 and amd64 on different platforms. I don’t think it hurts to keep it as is.

**Review by drexin — APPROVED** [review]

**Comment by millenomi** [comment]
Uh, wait, no. The CMake build isn't supported on Mac. What's up?

**Comment by DanboDuan** [comment]
> Uh, wait, no. The CMake build isn't supported on Mac. What's up?

but the `https://github.com/apple/swift` project needs it. when build the swift project with CMake and Ninjia, it will fail for arm64 not supported

**Comment by krzyzanowskim** [comment]
Hi, looks like this PR is approved for quite some time already. What blocks the merge?

**Comment by millenomi** [comment]
I'm okay with having arm64 support, but this isn't meant to be building _with CMake_ for Macs at all.

**Comment by millenomi** [comment]
Happy to merge if someone explains what's up here.

**Comment by DanboDuan** [comment]
> Happy to merge if someone explains what's up here.

this is for building `https://github.com/apple/swift` project with CMake and Ninja on M1 Mac. it will fail for arm64 not supported if arm64 is not supported @millenomi

**Pull request closed without merging**
